library/core/src/option.rs

```diff
@@ -70,10 +70,18 @@
 //! }
 //! ```
 //!
-//! This usage of [`Option`] to create safe nullable pointers is so
-//! common that Rust does special optimizations to make the
-//! representation of [`Option`]`<`[`Box<T>`]`>` a single pointer. Optional pointers
-//! in Rust are stored as efficiently as any other pointer type.
+//! # Representation
+//!
+//! Rust guarantees to optimise the following inner types such that an [`Option`] which contains
+//! them has the same size as a pointer:
```

Comment:
```suggestion
//! Rust guarantees to optimize the following types `T` such that an [`Option<T>`] 
//! has the same size as a `T`:
```
Giving names to things helps. :) And we typically use US spelling, AFAIK.

Also maybe say that this means that one may transmute a `T` to an `Option<T>`?

Reply:
Good point. Fixed in https://github.com/rust-lang/rust/pull/75454/commits/f5118a525fcf9db4102d903650331039158eff11.